tms/models/tms_order.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2024 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class TMSOrder(models.Model):
+    _name = "tms.order"
+    _description = "Transport Management System Order"
+
+    # _inherit = ["mail.thread", "mail.activity.mixin"]
+
+    active = fields.Boolean(default=True)
+    name = fields.Char(
+        required=True, index=True, copy=False, default=lambda self: _("New")
+    )
+    description = fields.Text()
+    sequence = fields.Integer(default=10)
+
+    route = fields.Boolean(string="Use predefined route")
+    route_id = fields.Many2one("tms.route")
+    origin_location = fields.Char(string="Origin")
+    destination_location = fields.Char(string="Destination")
+
+    driver_id = fields.Many2one("res.partner", string="Driver")
+    vehicle_id = fields.Many2one("tms.vehicle", string="Vehicle")
+    team_id = fields.Many2one("tms.team", string="Team")
+    crew_id = fields.Many2one("tms.crew", string="Crew")
+
+    stage_id = fields.Many2one(
+        "tms.stage",
+        string="Stage",
+        index=True,
+        copy=False,
+        default=lambda self: self._default_stage_id(),
+    )
+    stage = fields.Char(related="stage_id.name")
+
+    scheduled_date_start = fields.Datetime(
+        string="Scheduled Start", default=datetime.now()
+    )
+    scheduled_duration = fields.Float(
+        help="Scheduled duration of the work in" " hours",
+    )
+    scheduled_date_end = fields.Datetime(string="Scheduled End")
+
+    start_trip = fields.Boolean(readonly=True)
+    end_trip = fields.Boolean(readonly=True)
+    date_start = fields.Datetime(readonly=True)
+    date_end = fields.Datetime(readonly=True)
+    duration = fields.Char(readonly=True)
+
+    def _default_stage_id(self):
+        stage = self.env["tms.stage"].search(
+            [
+                ("stage_type", "=", "order"),
+                ("is_default", "=", True),
+                ("company_id", "in", (self.env.company.id, False)),
+            ],
+            order="sequence asc",
+            limit=1,
+        )
+        if stage:
+            return stage
+        raise ValidationError(_("You must create an TMS order stage first."))
+
+    @api.onchange("route")
+    def _onchange_route(self):
```

Comment:
To adopt current Odoo approach, change these onchanges to computed stored readonly=False fields instead.

Reply:
Can you send an example please?
What i'm trying to do is to empty the "route_id" field when I de-select the "route" boolean.

Reply:
Yes, that flow is supported.